File: app/src/main/java/com/example/movieapp102/viewmodel/MovieListViewModel.java
package com.example.movieapp102.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.movieapp102.model.Movie;
import com.example.movieapp102.repository.MovieRepository;

import java.util.List;

public class MovieListViewModel extends ViewModel {
    private MovieRepository movieRepository;

    public MovieListViewModel() {
        movieRepository = MovieRepository.getInstance();
    }

    public LiveData<List<Movie>> getMovies() {
        return movieRepository.getMovies();
    }

    public void searchMovieApi(String query, int page) {
        movieRepository.searchMovieApi(query, page);
    }
}

File: app/src/main/java/com/example/movieapp102/viewmodel/MovieListViewModel.kt
package com.example.movieapp102.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.movieapp102.model.Movie
import com.example.movieapp102.repository.MovieRepository

class MovieListViewModel : ViewModel() {

	private val movieRepository: MovieRepository = MovieRepository.getInstance()

	fun getMovies(): LiveData<List<Movie>> {
		return movieRepository.getMovies()
	}

	fun searchMovieApi(query: String) {
		movieRepository.searchMovie(query)
	}
}